shared/lib/accounts/snaps.ts

```diff
@@ -21,3 +37,43 @@ const WHITELISTED_SNAPS = [BITCOIN_WALLET_SNAP_ID, SOLANA_WALLET_SNAP_ID];
 export function isMultichainWalletSnap(id: SnapId): boolean {
   return WHITELISTED_SNAPS.includes(id);
 }
+
+/**
+ * Check if a Snap is a preinstalled Snap.
+ *
+ * @param snapId - Snap ID to verify.
+ * @returns True if Snap is a preinstalled Snap, false otherwise.
+ */
+export function isSnapPreinstalled(snapId: SnapId) {
+  return PREINSTALLED_SNAPS.some((snap) => snap === snapId);
+}
+
+/**
+ * Get the localized Snap name or some fallback name otherwise.
+ *
+ * @param snapId - Snap ID.
+ * @param messenger - Snap keyring messenger.
+ * @returns The Snap name.
+ */
+export function getSnapName(
+  snapId: SnapId,
+  messenger: SnapKeyringBuilderMessenger,
+) {
+  const { currentLocale } = messenger.call('PreferencesController:getState');
+  const snap = messenger.call('SnapController:get', snapId);
+
+  if (!snap) {
+    return stripSnapPrefix(snapId);
+  }
+
+  if (snap.localizationFiles) {
+    const localizedManifest = getLocalizedSnapManifest(
+      snap.manifest,
+      currentLocale,
+      snap.localizationFiles,
+    );
+    return localizedManifest.proposedName;
+  }
+
+  return snap.manifest.proposedName;
+}
```

Comment:
Same here, we probably should move those 2 in `shared/lib/snaps.ts`

Reply:
Done